### PR TITLE
Addition of saveMeta node

### PR DESCRIPTION
**Pull request by Dianeod**
- Reviewed by Deanna

### CODE REVIEW

**Comment by Dianeod** [comment]
- Addition of saveMeta node

- Removed`pathDict` parameter from `pathConstruct` as it's already created in `dataCheck`

File: code/nodes/pathConstruct/pathConstruct.q [hunk 1]
```diff
@@ -14,8 +14,7 @@
 //  paths to where graphs/reports will be generated
 pathConstruct.node.function:{[preProcParams;predictionStore]
   paths:pathConstruct.constructPath[preProcParams];
```
Comment (cmccarthy1) [inline code/nodes/pathConstruct/pathConstruct.q]: no need to assign this now

**Comment by Dianeod** [comment]
- All comments resolved

- Addition of tests in saveMeta to ensure correct data is being saved down

- Addition of tests in pathConstruct to ensure correct directories are being created

File: code/nodes/saveMeta/tests/test.t [hunk 2]
```diff
@@ -1 +1,64 @@
...
+
+passingTest[metaCheck;(paramDict0;savePath);0b;"No metadata"]
+passingTest[metaCheck;(paramDict1;savePath);0b;raze paramDict1]
+passingTest[metaCheck;(paramDict2;savePath);0b;raze paramDict2]
+
+-1"\nRemoving any directories created";
+
+// Remove any directories made
+rmPath:.automl.utils.ssrwin .automl.path,"/outputs/testing/";
+system"rm -r ",rmPath;
```
Comment (cmccarthy1) [inline code/nodes/saveMeta/tests/test.t:64]: Windows deletion also needs to be supported

**Review by cmccarthy1 — CHANGES_REQUESTED** [review]

**Pull request merged**